rhino/src/main/java/org/mozilla/javascript/NativeMath.java

```diff
@@ -246,6 +248,152 @@ private static Object floor(Context cx, Scriptable scope, Scriptable thisObj, Ob
         return ScriptRuntime.wrapNumber(x);
     }
 
+    private static Object f16round(
+            Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        // Handle missing arguments
+        if (args.length == 0) {
+            return ScriptRuntime.NaNobj;
+        }
+        double x = ScriptRuntime.toNumber(args[0]);
+
+        // Handle special cases
+        if (Double.isNaN(x)) return ScriptRuntime.NaNobj;
+        if (x == 0.0) return ScriptRuntime.wrapNumber(x); // Preserve sign of zero
+        if (Double.isInfinite(x)) return ScriptRuntime.wrapNumber(x);
+
+        // Extract components from double precision
+        long bits = Double.doubleToLongBits(x);
+        int sign = (int) (bits >>> 63);
+        int exponent = (int) ((bits >>> 52) & 0x7FF);
+        long mantissa = bits & 0x000FFFFFFFFFFFFFL;
+
+        // Adjust from double bias (1023) to float16 bias (15)
+        exponent = exponent - 1023 + 15;
+
+        // Handle overflow to infinity
+        if (exponent >= 31) {
+            return ScriptRuntime.wrapNumber(
+                    (sign != 0) ? Double.NEGATIVE_INFINITY : Double.POSITIVE_INFINITY);
+        }
+
+        // Handle underflow and subnormal values
+        if (exponent < 0) {
+            return handleSubnormalF16(sign, exponent, mantissa);
+        }
+
+        // Normal value: round mantissa from 52 to 10 bits
+        return handleNormalF16(sign, exponent, mantissa);
+    }
+
+    private static Object handleSubnormalF16(int sign, int exponent, long mantissa) {
+        // Values below 2^-24 underflow to zero
+        if (exponent < -10) {
+            return ScriptRuntime.wrapNumber((sign != 0) ? -0.0 : 0.0);
+        }
+
+        // Special case: exactly 2^-25 rounds to zero (ties-to-even)
+        if (exponent == -10 && mantissa == 0) {
+            return ScriptRuntime.wrapNumber((sign != 0) ? -0.0 : 0.0);
+        }
+
+        // Special case: slightly above 2^-25 rounds to 2^-24
+        if (exponent == -10 && mantissa > 0) {
+            double smallestSubnormal = 5.960464477539063e-8; // 2^-24
+            return ScriptRuntime.wrapNumber((sign != 0) ? -smallestSubnormal : smallestSubnormal);
```

Comment:
@anivar I get a compiler warning here:
```
/home/roland/dev/rhino/rhino/src/main/java/org/mozilla/javascript/NativeMath.java:301: warning: [FloatingPointLiteralPrecision] Floating point literal loses precision
            double smallestSubnormal = 5.960464477539063e-8; // 2^-24
                                       ^
    (see https://errorprone.info/bugpattern/FloatingPointLiteralPrecision)
  Did you mean 'double smallestSubnormal = 5.9604644775390625E-8;'?
1 warning
```
I would suggest to do this equivalent to `Double.MIN_NORMAL = 0x1.0p-1022` and can declare it as `smallestSubnormal = 0x1.0p-24`